redbot/cogs/mod/kickban.py

```diff
@@ -330,9 +330,13 @@ def remove_processed(ids):
         for entry in ban_list:
             for user_id in user_ids:
                 if entry.user.id == user_id:
-                    errors[user_id] = _("User {user_id} is already banned.").format(
-                        user_id=user_id
-                    )
+                    if user_id in await self.config.guild(guild).current_tempbans():
```

Comment:
You should get the list of tempbans *before* looping so that we can just reuse it instead of calling for it on each iteration.

Reply:
You're absolutely right... I thought I needed those for the user IDs specifically (don't ask me how that got into my brain). I put them up top to where the other local variables are defined.